src/libpmemobj/ctl.c

```diff
@@ -95,6 +95,14 @@ ctl_find_node(struct ctl_node *nodes, const char *name,
 	while (node_name != NULL) {
 		char *endptr;
 		long index_value = strtol(node_name, &endptr, 0);
+#ifdef __FreeBSD__
+/*
+ * FreeBSD returns EINVAL if no conversion performed, Linux does not.
+ * XXX - Other errors are possible, but are not checked.
```

Comment:
This code relies on checking the endptr value against the beginning of the string to check if the coversion has been performed, errno doesn't need to be checked - the higher level code will report that the entire string cannot be parsed.

I think we should just simply do:
```
int tmp = errno;
... strtol() ...;
errno = tmp;
```

That will be valid on all OSes.